modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -66,27 +81,39 @@ const setCursorInCell = (editor: Editor, cell: SugarElement<HTMLTableCellElement
  * - part of a table and content outside is selected
  */
 const emptySingleTableCells = (editor: Editor, cells: SugarElement<HTMLTableCellElement>[], outsideDetails: Optional<OutsideTableDetails>): boolean => {
-  // Remove content from selected cells
-  cleanCells(cells);
+  const editorRng = editor.selection.getRng();
+  const cellsToClean = outsideDetails.bind(({ rng, isStartInTable }) => {
+    /**
+     * Delete all content outside of the table that is in the selection
+     */
 
-  // Delete all content outside of the table that is in the selection
-  outsideDetails.map(({ rng, isStartInTable }) => {
     // Get the outside block before deleting the contents
     const outsideBlock = getOutsideBlock(editor, isStartInTable ? rng.endContainer : rng.startContainer);
     rng.deleteContents();
     // Handle block outside the table if it is empty since rng.deleteContents leaves it
     handleEmptyBlock(editor, isStartInTable, outsideBlock.filter(Empty.isEmpty));
-  });
 
-  // Set the selection:
-  // - to the first emptied cell if the start of the previous selection was inside a table
-  // - otherwise just collapse the previous selection that started in the outside block
-  if (outsideDetails.forall((details) => details.isStartInTable)) {
-    setCursorInCell(editor, cells[0]);
-  } else {
-    editor.selection.collapse(true);
-  }
+    /**
```

Comment:
```suggestion
    /*
```
As above, 2 asterisks are for API docs comments, a single one is just for a general multiline comment.